### PR TITLE
Complements of FiniteSet with Symbols returned unevaluated

### LINKED ISSUE #9808: Complements with symbols should remain unevaluated
Complements with symbols should remain unevaluated for the same reason Intersections with symbols should remain unevaluated i.e., we don't know what the complement is unless we the value of the symbol. Our current behaviour is:
```python
In [14]: Complement(FiniteSet(y), FiniteSet(1))
Out[14]: {y} # should be {y}\{1}
```

### PR DESCRIPTION
**Pull request by gxyd**
fixed #9808

### CODE REVIEW

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -1784,6 +1784,12 @@ def _complement(self, other):
             elif nums == []:
                 return None
 
+        elif isinstance(other, FiniteSet):
+            elms_unkown = FiniteSet(*[el for el in self if other.contains(el) not in (True, False)])
```
Comment (jksuom) [inline sympy/sets/sets.py]: unknown?

Comment (gxyd) [inline sympy/sets/sets.py]: Elements in `self` which return `Contains` object for `other.contains`. Does not the name sound right for that ?

Comment (jksuom) [inline sympy/sets/sets.py]: spelling...

Comment (gxyd) [inline sympy/sets/sets.py]: Oh i see i mis-typed two times the same word.

**Comment by gxyd** [comment]
Are the tests-failures related to this ?

File: sympy/sets/sets.py [hunk 2]
```diff
@@ -1784,6 +1784,12 @@ def _complement(self, other):
             elif nums == []:
                 return None
 
+        elif isinstance(other, FiniteSet):
+            elms_unknown = FiniteSet(*[el for el in self if other.contains(el) not in (True, False)])
+            if elms_unknown == self:
```
Comment (asmeurer) [inline sympy/sets/sets.py]: Is this case tested?

Comment (gxyd) [inline sympy/sets/sets.py]: Yes, i checked using the `coverage-report` which showed it was tested.

**Comment by gxyd** [comment]
@shivamvats @asmeurer do you think this is the right fix ?

File: sympy/series/order.py [hunk 3]
```diff
@@ -415,6 +416,10 @@ def _eval_subs(self, old, new):
                         from sympy.solvers.solveset import solveset
                         d = Dummy()
                         sol = solveset(old - new.subs(var, d), d)
+                        if isinstance(sol, Complement):
```
Comment (hargup) [inline sympy/series/order.py]: Are these changes tested?

Comment (gxyd) [inline sympy/series/order.py]: Yes, this is already tested by `two tests`. One of them being `sin(1/x).series(x, oo, n=5)`, in the `sympy.core.tests.test_wester.py` .

Comment (hargup) [inline sympy/series/order.py]: Cool, can you unxfail those tests from `test_wester`.

Comment (gxyd) [inline sympy/series/order.py]: @hargup Sorry, there is a confusion between us. Those tests were already `not` having `@XFAIL`. My first commit failed some of the tests, related to the `series module` raising an error of `zip a non-interable`. My second commit fixes that problem present in my first commit. You can look those failing `tests` in the first commit, `travis-build`.

Comment (hargup) [inline sympy/series/order.py]: got it.

**Comment by hargup** [comment]
+1

**Comment by gxyd** [comment]
Thanks

**Pull request merged**